hedera-node/src/main/java/com/hedera/services/txns/crypto/validators/AllowanceChecks.java

```diff
@@ -103,49 +94,238 @@ ResponseCodeEnum validateNftAllowances(final List<NftAllowance> nftAllowancesLis
 	 * 		Account of the payer for the allowance approve/adjust txn.
 	 * @param maxLimitPerTxn
 	 * 		max allowance limit per transaction
+	 * @param view
+	 * 		working view
 	 * @return response code after validation
 	 */
-	default ResponseCodeEnum allowancesValidation(final List<CryptoAllowance> cryptoAllowances,
+	public ResponseCodeEnum allowancesValidation(final List<CryptoAllowance> cryptoAllowances,
 			final List<TokenAllowance> tokenAllowances,
 			final List<NftAllowance> nftAllowances,
 			final Account payerAccount,
-			final int maxLimitPerTxn) {
+			final int maxLimitPerTxn,
+			final StateView view) {
 
 		// feature flag for allowances
 		if (!isEnabled()) {
 			return NOT_SUPPORTED;
 		}
+
 		var validity = commonChecks(cryptoAllowances, tokenAllowances, nftAllowances, maxLimitPerTxn);
 		if (validity != OK) {
 			return validity;
 		}
 
-		validity = validateCryptoAllowances(cryptoAllowances, payerAccount);
+		final var accountStore = new AccountStore(validator, dynamicProperties, view.asReadOnlyAccountStore());
+		validity = validateCryptoAllowances(cryptoAllowances, payerAccount, accountStore);
 		if (validity != OK) {
 			return validity;
 		}
-
-		validity = validateFungibleTokenAllowances(tokenAllowances, payerAccount);
+		final var tokenStore = new TypedTokenStore(accountStore,
```

Comment:
Passing multiple `null`s to the constructor here looks confusing...maybe for now we can just have a second `TypedTokenStore` constructor that takes only the first 3 arguments (and sets the unused fields to `null` internally).

This suggests we may want a `ReadOnlyTokenStore` base class where `TypedTokenStore` is a subclass that also supports mutations. 🤔

Reply:
Added `ReadOnlyTokenStore` only with the accountStore, tokens, uniqueTokens, tokenRelns maps and added all `load*`methods to it. 
`TypedTokenStore` has all `commit* ` and `persist*` that mutates merkle types. 

Also used `ReadOnlyTokenStore` in `AllowanceChecks`